src/Microsoft.SqlTools.ServiceLayer/Migration/Helper/ExtensionMethods.cs

```diff
@@ -0,0 +1,31 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.SqlServer.DataCollection.Common.Contracts.OperationsInfrastructure;
+
+namespace Microsoft.SqlTools.ServiceLayer.Migration.Helper
+{
+    internal static class ExtensionMethods
+    {
+        public static void AddExceptions(this IDictionary<string, IEnumerable<ReportableException>> exceptionMap1, IDictionary<string, IEnumerable<ReportableException>> exceptionMap2)
+        {
+            foreach (var keyValuePair2 in exceptionMap2)
+            {
+                // If the dictionary already contains the key then merge them
+                if (exceptionMap1.ContainsKey(keyValuePair2.Key))
+                {
+                    foreach (var value in keyValuePair2.Value)
+                    {
+                        exceptionMap1[keyValuePair2.Key].Append(value);
```

Comment:
Does this not give an error? I tried this and it said IEnumerable doesn't have a definition of Append.

Reply:
i'm not seeing any build issues. Did you have build errors or run time errors?

Reply:
For a similar change of dictionary check in the NuGet side, I was just trying this change out in a separate console app and this was what I got, hence I used Concat:

![image](https://user-images.githubusercontent.com/65304807/199113870-8de4c56e-5d35-410e-b72e-86485c78e82d.png)

If it is working on your end then that's good, maybe I am missing something.

Reply:
hmm i see. not seeing any errors on my end: 
![image](https://user-images.githubusercontent.com/9324681/199129446-12706e10-6da5-4447-a793-946088d20a2c.png)

not sure where the discrepancy comes from. i'll try running an end to end test to make sure it's working.

Reply:
tested this and it is working